qvisor/src/kvm_vcpu_x86.rs

```diff
@@ -398,6 +401,18 @@ impl KVMVcpu {
                         qlib::HYPERCALL_RELEASE_VCPU => {
                             SyncMgr::WakeShareSpaceReady();
                         }
+
+                        qlib::HYPERCALL_SHARESPACE_INIT => {
+                            let vms = VMS.lock();
```

Comment:
Dead lock happens here (I tested it). So please get the value and release the lock